lint-rules/src/main/java/com/ichi2/anki/lint/rules/CopyrightHeaderChecker.java

```diff
@@ -0,0 +1,100 @@
+/*
+ *  Copyright (c) 2021 David Allison <davidallisongithub@gmail.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify it under
+ *  the terms of the GNU General Public License as published by the Free Software
+ *  Foundation; either version 3 of the License, or (at your option) any later
+ *  version.
+ *
+ *  This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ *  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ *  PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with
+ *  this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.ichi2.anki.lint.rules;
+
+import com.android.tools.lint.detector.api.Context;
+import com.android.tools.lint.detector.api.Detector;
+import com.android.tools.lint.detector.api.Implementation;
+import com.android.tools.lint.detector.api.Issue;
+import com.android.tools.lint.detector.api.Scope;
+import com.android.tools.lint.detector.api.SourceCodeScanner;
+import com.google.common.annotations.VisibleForTesting;
+import com.ichi2.anki.lint.utils.Constants;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.uast.UClass;
+import org.jetbrains.uast.UElement;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CopyrightHeaderChecker extends Detector implements SourceCodeScanner {
+    @VisibleForTesting
+    static final String ID = "CopyrightHeaderChecker";
+    @VisibleForTesting
+    static final String DESCRIPTION = "Add Copyright Header in each file";
+    private static final String EXPLANATION = "All files in AnkiDroid must contain  Copyright Header. " +
```

Comment:
You might want to add newlines here as well.

```suggestion
    private static final String EXPLANATION = "All files in AnkiDroid must contain a GPLv3-compatible copyright header" +
```

You should also provide a link to a "default" copyright header that someone can copy/paste (maybe a wiki page?).